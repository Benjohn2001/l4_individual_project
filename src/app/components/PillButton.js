import { Component } from "react";
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, View } from "react-native";
import { Feather } from '@expo/vector-icons';

class PillButton extends Component {
    render(){
        const { title, onPress, icon} = this.props;
		return (
            <View className="items-center py-3">
                <TouchableOpacity onPress={onPress} className="bg-darkerPurple h-8 w-44 justify-between rounded-full mx-1.5">
                    <View className="flex-row justify-center">
                        <View className="pt-2">
                            <Feather name={icon} size={16} color="white" />
                        </View>
                        <View className="pt-1">
                            <Text  className="text-white text-sm text-center ml-5">{title}</Text>
                        </View>
                    </View>
                </TouchableOpacity>
            </View>
        );
    }
}

PillButton.propTypes = {
    title: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    icon: PropTypes.string.isRequired
};

export default PillButton;