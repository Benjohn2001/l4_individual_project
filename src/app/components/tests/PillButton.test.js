import { render, screen, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import PillButton from "../PillButton";

test("Check PillButton renders correctly and onpress works", () => {
  const onPressMock = jest.fn();
  render(
    <PillButton
      title="press"
      onPress={() => {
        onPressMock();
      }}
      icon="users"
    />
  );
  fireEvent.press(screen.getByText("press"));
  expect(onPressMock);
  const domTree = renderer
    .create(
      <PillButton
        title="press"
        onPress={() => {
          onPressMock();
        }}
        icon="users"
      />
    )
    .toJSON();
  expect(domTree).toMatchSnapshot();
});
