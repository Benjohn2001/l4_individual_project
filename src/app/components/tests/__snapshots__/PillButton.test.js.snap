// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check PillButton renders correctly and onpress works 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6B4EFF",
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
        "height": 32,
        "justifyContent": "space-between",
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
        "width": 176,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text />
      </View>
      <View
        style={
          Array [
            Object {
              "paddingTop": 4,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
              },
              Object {
                "fontSize": 14,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "marginLeft": 20,
              },
            ]
          }
        >
          press
        </Text>
      </View>
    </View>
  </View>
</View>
`;
